mesonbuild/interpreter/interpreterobjects.py

```diff
@@ -229,76 +234,52 @@ def stdout_method(self, args: T.List[TYPE_var], kwargs: TYPE_kwargs) -> str:
     def stderr_method(self, args: T.List[TYPE_var], kwargs: TYPE_kwargs) -> str:
         return self.stderr
 
-# TODO: Parsing the initial values should be either done directly in the
-#       `Interpreter` or in `build.EnvironmentVariables`. This way, this class
-#       can be converted into a pure object holder.
-class EnvironmentVariablesObject(MutableInterpreterObject, MesonInterpreterObject):
-    # TODO: Move the type cheking for initial_values out of this class and replace T.Any
-    def __init__(self, initial_values: T.Optional[T.Any] = None, subproject: str = ''):
-        super().__init__(subproject=subproject)
-        self.vars = build.EnvironmentVariables()
+
+_ENV_SEPARATOR_KW = KwargInfo('separator', str, default=os.pathsep)
+
+
+class EnvironmentVariablesObject(ObjectHolder[build.EnvironmentVariables], MutableInterpreterObject):
```

Comment:
For consistency, this should be called `EnvironmentVariablesHolder`

Reply:
yeah, that makes sense.